pylint/lint/pylinter.py

```diff
@@ -291,12 +292,18 @@ def __init__(
         options: Options = (),
         reporter: reporters.BaseReporter | reporters.MultiReporter | None = None,
         option_groups: tuple[tuple[str, str], ...] = (),
-        # TODO: Deprecate passing the pylintrc parameter
-        pylintrc: str | None = None,  # pylint: disable=unused-argument
+        pylintrc: str | None = None,
```

Comment:
Should we use a sentinal `object` as default value so that we can explicitly check whether people are passing `None` here?

Reply:
Right, I wasn't sure whether to bother?